pkg/agent/client.go

```diff
@@ -289,11 +295,19 @@ func (a *Client) Recv() (*client.Packet, error) {
 	return pkt, err
 }
 
-func serverCount(stream agent.AgentService_ConnectClient) (int, error) {
+func getAndValidateStreamHeader(stream agent.AgentService_ConnectClient) (metadata.MD, error) {
+	// TODO: this is a blocking call. Add a timeout?
 	md, err := stream.Header()
```

Comment:
For the reader / talking to myself: this change adds a new cleaner 403 signal from server to agent during bootstrapping. So it's natural to worry about the new agent binary / old server binary scenario, and the possibility that the agent will get stuck waiting for this new signal from the server. But, the old agent already blocked on .Header() in the same place as before (serverCount and serverID) so this alone isn't a risk.